File: smart-spin-api/src/main/java/com/example/smartspinapi/model/exception/UserIsNotMultiplayerQuizLeaderException.java
package com.example.smartspinapi.model.exception;

public class UserIsNotMultiplayerQuizLeaderException extends Exception {
    public UserIsNotMultiplayerQuizLeaderException(String userId, String joinCode) {
        super("User with id " + userId + " is not the leader of " + joinCode);
    }
}

File: Frontend/trivia_app/lib/core/domain/exceptions/user_is_not_multiplayer_quiz_leader_exception.dart
class UserIsNotMultiplayerQuizLeaderException implements Exception {
  UserIsNotMultiplayerQuizLeaderException(this.statusCode, {this.message});

  final int statusCode;
  final String? message;

  @override
  String toString() {
    return message ?? '"User is not the leader $statusCode';
  }
}
